### DISCUSSION TITLE
Is there any update of TR recently

### DISCUSSION THREAD

**Discussion by friendfish** [thinkingrock-gtd/tr-pc#12] [General]
Hi guys, I am a big fan of TR and always try to find new version of TR.
finally catch the project here, is there any road map for TR

**Comment by ursjoss** [comment]
Hi @friendfish Thanks for your interest. Unfortunately, we're far from having a road map. After a brief storming phase, we're more or less stuck in the forming phase (to stress tuckman's group development model). We have two pull requests as a first start to get some stuff done, but so far, we haven't been able to merge them due to a lack of time or skill to test and review them. Do you have some technical skills and enough enthusiasm to get involved?

**Reply by johandersson** [reply]
I echo @friendfish love for TR. I saw that you changed the license, is the plan, since you put it in a public repo, that TR is going to be free software completely or still have a license fee? I tried to build the project in Netbeans but got an error. I would also like to try to see if I could help in some way.

**Reply by ursjoss** [reply]
@johandersson The idea was/is to open-source TR. Currently, we're lacking a few roles to actually go on. I created a couple of pull requests, but I'm hesitant to merge them before some additional eyes review the changes and the effect. If you (and/or @RadoslavCap ) can jump in to do code review and/or test the branches, we could actually merge the PRs and go on with additional topics. Until then, I am hesitant to tackle additional topics.

If you want to try a working state for now, use the main branch of my [fork](https://github.com/ursjoss/tr-pc). I merged all currently open PRs in that forked repo, so I can start "dogfooding". I'm using a TR instance built from current main personally.

Beware: The open-sourced version of the TR is currently still lacking the settings module.

so yes: we are very happy if more people would start to dig in and contribute with whatever skills they have.

**Reply by johandersson** [reply]
Thanks for the reply.

Now I could build your fork in Netbeans, but when running I get the following error:

SEVERE: No way to find original stream handler for jar protocol
java.lang.reflect.InaccessibleObjectException: Unable to make field transient java.net.URLStreamHandler java.net.URL.handler accessible: module java.base does not "opens java.net" to unnamed module @2a26551a
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:180)
	at java.base/java.lang.reflect.Field.setAccessible(Field.java:174)
	at org.netbeans.ProxyURLStreamHandlerFactory.register(ProxyURLStreamHandlerFactory.java:59)
	at org.netbeans.JarClassLoader.<clinit>(JarClassLoader.java:117)
	at org.netbeans.MainImpl.execute(MainImpl.java:153)
	at org.netbeans.MainImpl.main(MainImpl.java:60)
	at org.netbeans.Main.main(Main.java:58)

I use Java 18 and OpenJDK and Ubuntu Linux.

**Reply by ursjoss** [reply]
> Thanks for the reply.
> 
> Now I could build your fork in Netbeans, but when running I get the following error:
> 
> I use Java 18 and OpenJDK and Ubuntu Linux.

That state of the `main` branch is tested with and for Java 11 only. It will be one of my first steps to go all the way to Java 17 (see #17) as the current LTS, but likely only after we merge the existing PRs. Chances are probably not too bad that that would also work with Java 18, but I don't know.

Not reconciled with anyone else's oppinion, but here's the issues in the [project board](https://github.com/thinkingrock-gtd/tr-pc/projects/1).

**Reply by johandersson** [reply]
Thanks,

Now I could run it in Netbeans with Java 11. What would be a good small task to start to help?

**Reply by ursjoss** [reply]
Great news, @johandersson , thanks for staying with it...

If you can look at the open pull requests (maybe starting with #8 ) and provide your experience and findings for those. If all is good, we can merge them and get on with additional topics.

you should be able to find all relevant information in the PR. Don't hesitate to come back here if you need more info. I'll be glad to assist.

**Reply by degrown** [reply]
Hello @johandersson , I can confirm what @ursjoss said before.
In particular:
- I was a Python programmer (no Java experience), and now I'm managing people...so @ursjoss waited for me for too long, but technically I'm not able to help him. So I'm welcoming your and @RadekCap contributions. We waited a lot for someone to show up, so...welcome again!
- opensource: yes, the plan is to go opensource, but the first point is to be alive, so: be able to start this application from a current OS (I'm a macOS user, and I'm very very worried about that) 
- about the project board: thank you @ursjoss to handle that, a kanban work view is really a must with more than one people!

I can also say that: I'll do whatever I can to help this project thrive again; I've tried for few years, maybe we can proceed with your help

**Reply by salariua** [reply]
Hi all,

I came here from the trgrd forum and I am willing to help with any testing you might have but sadly I have no programming experience. I am a mechanical design engineer by day so not much help here :(

Thanks,
Adrian

**Reply by ursjoss** [reply]
Hi Adrian

Thanks for coming here. We are still in a relatively early phase, getting to know the code base, we have fixed some bugs and are in the process of applying some houe-keeping in the code base.

Can we reach out to you once we get a bit closer to a first release? Then we will ceratinly need all the testing we can get. Or how would you see your potential contribution in the mean-time?

Best regards
Urs

**Comment by degrown** [comment]
Hello @friendfish , I confirm @ursjoss words: we currently don't have a roadmap, also because we need people who can help us on technical side. Would you like to try?
Thanks!

**Reply by RadekCap** [reply]
Hmmm, I might give it a try.

**Reply by ursjoss** [reply]
Hi @RadoslavCap 

We could certainly use your skill sets :-). @degrown Would you be able to add Radoslav to our internal discussion forum so we can discuss the next steps?

Thanks

**Comment by ursjoss** [comment]
Hi @RadoslavCap, @johandersson , @friendfish 

You have asked for plans or hinted that you'd might be able to find some time to contribute to the next steps of thinking rock. I'd highly appreciate it if somebody would take the time to

- review pull requests for issues I've tackled and/or
- test the functionality of the changes.

Please have a look at the [project board](https://github.com/thinkingrock-gtd/tr-pc/projects/1).

In my own fork of the project, I have already merged all open PRs from this project and I'm using this version of thinking rock productively with java 17. It allows me to export PDFs, use email/POP to import issues etc. Most of my kinda blocking issues are solved for me. It would be time to start merging the PRs. We should tackle them one after the other, as some of the subsequent PRs include the previous ones as well.  By merging them in order, we can keep the merged changes more cohesive.

Please let me know if you'd like to start with reviews/tests but are blocked by something. I'm more than glad helping out to help additional contributers get to speed.

**Reply by RadekCap** [reply]
Hmmm, @ursjoss looks like I'm a little bit blocked by my knowledge of Netbeans and Ant. 

I'm able to compile it with BUILD_SUCCESFULL on RHEL. I have installed TR on the same machine as well. During compilation, I hit the issue with goals that aren't probably supported yet by our sources. I created an empty TR file without goals and the compilation ran successfully. But it rewrites my ReviewActions.xml file which I'm using with my installed application (!). So it's probably somehow interfering with the installed TR on my machine. etc. etc.

As you can see there are a lot of assumptions and guesses so maybe a short call allows me not to focus on basic things and focus on PR. WDYT?

**Reply by ursjoss** [reply]
@RadoslavCap Sure, I contacted you out-of-band to arrange a call...

**Reply by ursjoss** [reply]
> @RadoslavCap Sure, I contacted you out-of-band to arrange a call...

In case you didn't see that, I reached out on Linkedin.

**Reply by RadekCap** [reply]
Yep, I missed it. Now, I'm there.

**Comment by salariua** [comment]
Hi Urs,

Whatever works for you guys.

Can't think of what use you might have for me at the moment but thought I
start by making an account and get in touch :)

Thanks,
Adrian

**Reply by ursjoss** [reply]
Great to know, Adrian. I have not much time today, but I'll think about how we could include you for some test runs...

**Reply by ursjoss** [reply]
@salariua on what OS do you run ThinkingRock?

**Reply by salariua** [reply]
@ursjoss I am running TR on windows 10 Pro.

**Reply by degrown** [reply]
Thank you so much @salariua for being willing to help!
We really need everyone for TR to strive again; as @ursjoss wrote, maybe you'll have to wait a little until we organize windows tests..but testing is for sure an important activity (for example, I'm on macOS, so your OS will be a good candidate for different setups)

Thank you again, and stay tuned!
